---
int8[2] state